proposed/2021/NuGet.orgTFMs.md

```diff
@@ -0,0 +1,94 @@
+# NuGet Target Frameworks on Package Details Page
+
+- Author Name (https://github.com/jcjiang)
+- Start Date (2021-02-16)
+- GitHub Issue (https://github.com/NuGet/NuGetGallery/issues/4843)
+- GitHub PR ()
+
+## Summary
+
+When a developer opens the package details page or tab on NuGet.org, they will see what frameworks the package can support. 
+
+## Motivation
+
+### Problem: What problem is this solving? 
+
+Developers cannot use the package details page to determine what frameworks a NuGet package can support. In 2021, we live in a world where there are many evolutions of .NET coexisting on NuGet, such as .NET Framework and .NET 5.
+
+Developers should be able to determine whether a package works for their project without having to first download it or rely on the author’s documentation. 
+
+### Why: How do we know this is a real problem and worth solving? 
+
+Finding a package that is compatible with your project is a challenge for the average developer in the .NET ecosystem. Over 21% of developers fail to install a package today. One of the top NuGet errors issued in telemetry is [NU1202](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1202), which shows that we allow developers to install packages that are not compatible with their project or solution’s target framework. 
+
+### Success: How do we know if we’ve solved this problem? 
+
+- By gauging excitement/disappointment after blogging about this feature & analyzing sentiment on Twitter/GitHub/DevCom/etc. 
+
+### Audience: Who are we building for? 
+
+We are building this for the .NET developer who is browsing & making decisions about including a NuGet package into their project based on target framework & supported platform criteria. 
+
+## Explanation
+
+### Functional Explanation
+
+Context and Scope: When a developer accesses the details page for a NuGet package, they will see information on target frameworks in an expandable section. The framework information will be surfaced from the package.
+
+Minimal Requirements:
+- List of Target Frameworks on NuGet.org's package details page.
+- Runtime, Content Build assets and Tools packages will display TFM details.
+- Support for PackageReference project restore compatibility only.
+
+**Target Framework** – A target framework is a specification of the APIs available to an application or library. 
+
+Example: .NET Framework, .NET Core, .NET Standard, .NET 5 are target frameworks. 
+
+**Supported Platform** – A supported platform is a target that the APIs can be ran on. 
+
+Example: iOS, Android, Windows are platforms. 
+
+- A .NET developer can view Frameworks of a package on NuGet.org: 
+![](../../meta/resources/NuGet.orgTFMs/PackageDetailsWithTFMs.png)
+
+- We will have the list collapsed by default in order to track popularity.
```

Comment:
At an earlier point in time (about ~1 year ago), we did some analysis on `what frameworks does a package really support` vs `what frameworks are declared in the dependencies section` and we found that for about 95% of packages, these are a 1 to 1 relationship. 

Have we consider making that more apparent instead? In the interest of saving some vertical space/introducing a new concept that's already covered in most scenarios.

Reply:
This is a good point. There's a challenge here where you run into the XUnit case.

![image](https://user-images.githubusercontent.com/1288848/108431455-81ec8b80-7208-11eb-8b52-d6ab4e8b737b.png)

To get to an actual "idea" of if the package supports a specific framework is layers deep to the dependencies defined such as:

![image](https://user-images.githubusercontent.com/1288848/108431520-9597f200-7208-11eb-8c86-c5b520fb7522.png)

So in these cases, we don't have a good answer without iterating dependencies.

Reply:
That totally makes sense. 

I just thing integrating it differently, not in an additive way might save us some space. Right now there's *a lot*.

Reply:
We just need to consider a tab design for these pages as a "Future Possibility". `Dependencies`, `Frameworks`, `Version History`, `Dependents (Used By)`, etc really belong in their own individual tabs so everything isn't stacked ontop of each other.

Reply:
Discussed offline and we'll consider these to be universally compatible at this stage. I've added `- A design for meta packages` to future possibilities, and splitting expanders out to tabs may be called out here or in anothr spec. Up to you, @jcjiang. As we've nailed down the pertinent corner cases, for now perhaps we can merge this?